rust/src/accuracy/polars/mod.rs

```diff
@@ -0,0 +1,203 @@
+use opendp_derive::bootstrap;
+use polars::{
+    datatypes::{AnyValue, DataType, Field},
+    frame::{row::Row, DataFrame},
+    prelude::{IntoLazy, LazyFrame, Schema},
+};
+use polars_plan::{
+    dsl::{AggExpr, Expr},
+    plans::DslPlan,
+};
+
+use crate::{
+    accuracy::{
+        discrete_gaussian_scale_to_accuracy, discrete_laplacian_scale_to_accuracy,
+        gaussian_scale_to_accuracy, laplacian_scale_to_accuracy,
+    },
+    core::{Measure, Measurement, Metric, MetricSpace},
+    domains::LazyFrameDomain,
+    error::Fallible,
+    measurements::{
+        expr_noise::{match_noise, Distribution, Support},
+        expr_report_noisy_max_gumbel::match_report_noisy_max_gumbel,
+    },
+    polars::{ExtractLazyFrame, OnceFrame},
+    transformations::expr_discrete_quantile_score::match_discrete_quantile_score,
+};
+
+#[cfg(feature = "ffi")]
+mod ffi;
+
+#[bootstrap(
+    name = "onceframe_measurement_utility",
+    features("contrib"),
+    arguments(
+        measurement(rust_type = "AnyMeasurement"),
+        alpha(c_type = "AnyObject *", default = b"null")
+    ),
+    generics(MI(suppress), MO(suppress)),
+    returns(c_type = "FfiResult<AnyObject *>")
+)]
+/// Get noise scale parameters from a measurement that returns a OnceFrame.
+///
+/// # Arguments
+/// * `measurement` - computation from which you want to read noise scale parameters from
+/// * `alpha` - optional statistical significance to use to compute accuracy estimates
+pub fn onceframe_measurement_utility<MI: Metric, MO: 'static + Measure>(
+    measurement: Measurement<LazyFrameDomain, OnceFrame, MI, MO>,
+    alpha: Option<f64>,
+) -> Fallible<DataFrame>
+where
+    (LazyFrameDomain, MI): MetricSpace,
+{
+    let schema = measurement.input_domain.schema();
+    let lf = DataFrame::from_rows_and_schema(&[], &schema)?.lazy();
+    let mut of = measurement.invoke(&lf)?;
+    let lf: LazyFrame = of.eval_internal(&ExtractLazyFrame)?;
+
+    lazyframe_utility(&lf, alpha)
+}
+
+struct UtilitySummary {
+    pub name: String,
+    pub aggregate: String,
+    pub distribution: Option<String>,
+    pub scale: f64,
+    pub accuracy: Option<f64>,
+}
+
+/// Get noise scale parameters from a LazyFrame.
+///
+/// # Arguments
+/// * `lazyframe` - computation from which you want to read noise scale parameters from
+/// * `alpha` - optional statistical significance to use to compute accuracy estimates
+pub fn lazyframe_utility(lazyframe: &LazyFrame, alpha: Option<f64>) -> Fallible<DataFrame> {
+    let mut utility = logical_plan_utility(&lazyframe.logical_plan, alpha)?;
+
+    // only include the accuracy column if alpha is passed
+    if alpha.is_none() {
+        utility = utility.drop("accuracy")?;
+    }
```

Comment:
But called from python, I think alpha is always set? I guess I'm not sure why it needs to be optional.

Reply:
It isn't always set-- if alpha is not set, the function will just read off the distributions and scale parameters.